auto_tests/save_load_test.c

```diff
@@ -88,18 +88,43 @@ static void test_few_clients(void)
     ck_assert_msg(connected_t1, "Tox1 isn't connected. %u", connected_t1);
     printf("tox clients connected took %lu seconds\n", (unsigned long)(time(nullptr) - con_time));
 
+    /* validate that:
+     * a) saving stays within the confined space
+     * b) a saved state can be loaded back successfully
+     * c) a second save is of equal size
+     * d) the second save is of equal content */
+    const size_t extra = 64;
     const size_t save_size1 = tox_get_savedata_size(tox2);
     ck_assert_msg(save_size1 != 0, "save is invalid size %u", (unsigned)save_size1);
     printf("%u\n", (unsigned)save_size1);
-    VLA(uint8_t, save1, save_size1);
-    tox_get_savedata(tox2, save1);
+    VLA(uint8_t, buffer, save_size1 + 2 * extra);
```

Comment:
Perhaps while you're changing this code already, remove the vla and use malloc/free?